tutorial/pytorch_agent/train_cpu_model-pytorch.py

```diff
@@ -0,0 +1,336 @@
+        optimizer.step()
+
+    model.eval()
+    prediction_total = 0
+    prediction_correct = 0
+    val_loss = 0
+    val_steps = 0
+    with torch.no_grad():
+        for idx, (inputs, target_control) in enumerate(val_loader):
+            prediction_total += inputs.size(0)
+
+            # Run inputs through model, save prediction
+            predicted_control = model(inputs)
+            # Round to nearest 100 MHz increment
+            predicted_control = np.round(predicted_control,1)
+            prediction_correct += (target_control == predicted_control).sum().item()
+
+            loss = loss_fn(predicted_control, target_control)
+            val_loss += loss.numpy()
+            val_steps += 1;
+
+
+    return val_loss/val_steps, prediction_correct/prediction_total
+
+def test(model, test_loader):
+    prediction_total = 0
+    prediction_correct = 0
+    prediction_min = 0
+    prediction_max = 0
+    with torch.no_grad():
+        for idx, (inputs, target_control) in enumerate(test_loader):
+            prediction_total += inputs.size(0)
+            # Run inputs through model, save prediction
+            predicted_control = model(inputs)
+            # Round to nearest 100 MHz increment
+            predicted_control = np.round(predicted_control,1)
+
+            # Generate stats
+            prediction_correct += (target_control == predicted_control).sum().item()
+
+            if (idx == 0):
+                prediction_max = predicted_control.max()
+                prediction_min = predicted_control.min()
+            else:
+                if (prediction_max < predicted_control.max()):
+                    prediction_max = predicted_control.max()
+                if (prediction_min > predicted_control.min()):
+                    prediction_min = predicted_control.min()
+
+        print('\ttest output min: {:.2f}, max: {:.2f}.'.format(prediction_min, prediction_max))
+    return prediction_correct/prediction_total
+
+class P3Net(nn.Module):
+    def __init__(self, width_input, width_fc, depth_fc):
+        super(P3Net, self).__init__()
+        self.depth_fc = depth_fc
+        self.norm1 = nn.BatchNorm1d(width_input)
+        self.fc1 = nn.Linear(width_input, width_fc)
+        # This approach doesn't work for torchscript - torch.jit.script(best_model)
+        #self.fc_list = []
+        #for d in range(self.depth_fc):
+        #    self.fc_list.append(nn.Linear(width_fc, width_fc))
+        self.fc2 = nn.Linear(width_fc, width_fc)
+        self.sigmoid = nn.Sigmoid()
+        self.fc_out = nn.Linear(width_fc, 1)
+
+    def forward(self, x):
+        x = self.norm1(x)
+        x = self.fc1(x)
+        x = self.sigmoid(x)
+        # This approach doesn't work for torchscript - torch.jit.script(best_model)
+        #for d in range(self.depth_fc):
+        #    x = (self.fc_list[d])(x)
+        #    x = self.sigmoid(x)
+        for d in range(self.depth_fc):
+            x = self.fc2(x)
+            x = self.sigmoid(x)
```

Comment:
Using a list of fc layers does not seem to agree with torchscript.  We may have to chose between hyperparameter tuning & saving to a script model.  This would result in a two pass system (tune, then update the model depth & train with new settings for a model usable by the agent)

Reply:
Solved in commit bbe5cb2